main.go

```diff
@@ -779,6 +779,23 @@ func initialiseSystem(arguments map[string]interface{}) error {
 		afterConfSetup(&config.Global)
 	}
 
+	if os.Getenv("TYK_LOGLEVEL") == "" && arguments["--debug"] == false {
+		switch strings.ToLower(config.Global.LogLevel) {
+		case "", "info":
+			//defualt, do nothing
+		case "error":
+			log.Level = logrus.ErrorLevel
+		case "warn":
+			log.Level = logrus.WarnLevel
+		case "debug":
+			log.Level = logrus.DebugLevel
+		default:
+			log.WithFields(logrus.Fields{
+				"prefix": "main",
+			}).Fatal("Invalid log level in config, must be error, warn or debug.")
```

Comment:
or info

Reply:
also include the string, like `Fatalf("Invalid log level %q ...", level)`

you can reuse the var like:

```
level := strings.ToLower(config.Global.LogLevel)
switch level { ... }
```

Reply:
why %q and not %s in this case?

Reply:
Because `%q` will double-quote the string (see https://golang.org/pkg/fmt/)

So then, for `   confusing"` instead of `invalid log level    confusing", expected ...` you'll get `invalid log level "   confusing\"", expected...`.

Reply:
In practice both would work ok as I don't think anyone would put weird chars here. But we might as well make the fatal message more foolproof.